### ISSUE TITLE
Website wrapper (information)

### ISSUE THREAD

**Issue by pebll** [HliasOuzounis/Ai-Guess-the-elo#4]
Hey just wanted to let you know that a vibe coded an web wrapper for your super cool predictor: léo.com/elo-estimator

I wanted to build a model myself, but i did not have time (exams) so i made this instead. I found your readme very interesting, i might come back to it and try an alternative approach with transformers.

let me know what you think about it!

LG 

Léo

**Comment by pebll** [comment]
Also quick side question, I asked myself how your train data is distributed? Did you use a uniform distribution for all rating ranges (e.g. 1000 games per 50 ELO step) or did you just use all games from a particular month (gaussian around mean elo i guess 1600?). I tested it on many games of myself and on average (moving average) it very well predicts my elo :)

With Magnus Carlsen the goat however, it seems that the maximum the model can output is around 2200 elo, as it does with magnus consistently: 

<img width="1879" height="873" alt="Image" src="https://github.com/user-attachments/assets/1e733e7d-f5ab-4166-9b5c-32cff23b7690" />

**Comment by HliasOuzounis** [comment]
Hello Léo,

Thanks for checking out my project. Your website seems really cool and impressively well made. What backend did you use? When I finished this project I also had ideas about making a website but quickly gave up due to a lack of time.

This was a personal project I did as a University student as well. Obviously, this means that it has a lot of room for improvement, which I highly suggest you explore if you find it interesting. I consider it an interesting project that stands out from all the common "AI projects" that graduates like to show on interviews.

As you pointed out, the model is not very good at predicted high Elo games and constantly underestimates the players; the opposite happens for lower rated games. The nature of the problem definition with the normal distribution estimation, naturally leads to more predictions near the middle of the ratings ladder as they are "safer" in terms of the loss function. Couple this with the fact that there were fewer games for high rated players and you get the high discrepancy on the predictions for Magnus.

Some changes in the loss function to punish large differences more severely and more training data would certainly help. If you give it a shot, let me know of your results! I'd be interested to take a look.

**Comment by pebll** [comment]
Nice! Well i do not know lots of webdev myself, this was mostly vibe coded, i just wanted to get something out quickly and that was a nice little procrastination, and i wanted also mainly to learn how docker/backend/frontend/deployement ect works. The frontend is using flask (python) and this runs on my little personal VM with 1G ram that i got from my uni as a student. I also had a lecture this semester about transformers that gave me many reasons to use transformers on this task. I will 100% mention back if I make my own take on the model!
